### ISSUE TITLE
undefined symbol: cblas_dasum

### ISSUE THREAD

**Issue by boscotsang** [thenovices/heroku-buildpack-scipy#22]
After deployment the import error raised during runtime. The error report as follow:

ImportError: /app/.heroku/python/lib/python2.7/site-packages/sklearn/metrics/pairwise_fast.so: undefined symbol: cblas_dasum

**Comment by thenovices** [comment]
Can you provide more information about what versions of packages you're using?

There are known issues with installing sklearn on this buildpack with certain versions. I personally haven't had the time to look into sklearn in detail yet, since I don't use it very much.

**Comment by boscotsang** [comment]
My requirements.txt is as follow

numpy==1.9.2
scipy==0.15.1
scikit-learn==0.15.2
simplejson>=3.8.0
Flask>=0.10
Jinja2>=2.8
Werkzeug>=0.9
gunicorn>=0.17.2

**Comment by mitchelljkotler** [comment]
I am seeing this issue as well

**Comment by branliu0** [comment]
Is it also for sklearn specifically? I'm afraid I haven't been able to investigate sklearn issues with this buildpack.

**Comment by mitchelljkotler** [comment]
Yes, for sklearn.  Not having any luck getting sklearn going on heroku by
any means.

**Comment by mitchelljkotler** [comment]
Using this buildpack instead resolved the issue for me:
https://github.com/andrewychoi/heroku-buildpack-scipy

**Comment by WhiskySoda** [comment]
Still having this issue, tried the build pack... unfortunately, not much luck. sklearn==0.15.2

**Comment by 72L** [comment]
I'm also having this issue. Switching to https://github.com/andrewychoi/heroku-buildpack-scipy did not work for me scikit-learn==0.16.1, scipy==0.15.1, numpy==1.9.2

**Comment by anitameh** [comment]
I'm also having the same issue, even after switching to andrewychoi's buildpack. 

`ImportError: /app/.heroku/python/lib/python2.7/site-packages/sklearn/metrics/pairwise_fast.so: undefined symbol: cblas_dasum`

anybody manage to get this working?

**Comment by jdelman** [comment]
Also having this issue.

@mitchelljkotler and @thenovices - could you tell us which versions of sklearn, numpy, and scipy that you're using?

**Comment by branliu0** [comment]
Sorry guys -- I don't use sklearn at all, but I can try to look into this. I'm pretty sure we would have to make a new precompiled binary for sklearn as well. I don't have that much time right now to work on that -- but hopefully will get to it at some point

**Comment by jdelman** [comment]
Running using Choi's [buildpack](https://github.com/andrewychoi/heroku-buildpack-scipy) with the following versions:

```
numpy==1.9.1
scipy==0.14.0
scikit-learn==0.15.2
```

**Comment by mitchelljkotler** [comment]
Also with Choi's buildpack:
numpy==1.9.1
scipy==0.14.0
scikit-learn==0.16.1

**Comment by anitameh** [comment]
sweet got this working with choi's buildpack too! thanks!!

**Comment by seyonv** [comment]
I'm having the same problem. (Using Choi's build pack and versions specified by mitchelljkotler

 Getting the following error 
"ImportError: liblapack.so.3: cannot open shared object file: No such file or directory"

**Comment by troyshu** [comment]
@seyonv @mitchelljkotler's version numbers just worked for me. Make sure numpy, scipy, and scikit-learn are the very first things in your requirements.txt; also, I'm on Python 2.7.

**Comment by alexcombessie** [comment]
Same issue here. I solved it with Choi's build pack and the following versions on my requirements.txt:

```
numpy==1.9.1
scipy==0.14.0
scikit-learn==0.16.1
```

Be careful to reset all your buildpacks beforehand to do a clean install. You can do `heroku buildpacks:clear`

Just to be sure, I also `pip` uninstalled everything with `pip freeze | xargs pip uninstall -y`.

**Comment by FreshDonut** [comment]
Following all the tips. I feel like I'm getting closer. I do require `pandas` to read from a csv file though. I seem to be getting the following error on app load `ImportError: C extension: liblapack.so.3: cannot open shared object file: No such file or directory not built. If you want to import pandas from the source directory, you may need to run 'python setup.py build_ext --inplace' to build the C extensions first.` Is this bc it can't find the file from S3 or something to do with pandas? Any thoughts guys?
